cmd/xl-v1-object.go

```diff
@@ -969,11 +970,68 @@ func (xl xlObjects) ListObjectsV2(ctx context.Context, bucket, prefix, continuat
 	return listObjectsV2Info, err
 }
 
-// Send the successul but partial upload, however ignore
+// Send the successful but partial upload, however ignore
 // if the channel is blocked by other items.
 func (xl xlObjects) addPartialUpload(bucket, key string) {
 	select {
 	case xl.mrfUploadCh <- partialUpload{bucket: bucket, object: key}:
 	default:
 	}
 }
+
+// PutObjectTag - replace or add tags to an existing object
+func (xl xlObjects) PutObjectTag(ctx context.Context, bucket, object string, tags string) error {
+	disks := xl.getDisks()
+
+	// Read metadata associated with the object from all disks.
+	metaArr, errs := readAllXLMetadata(ctx, disks, bucket, object)
+
+	_, writeQuorum, err := objectQuorumFromMeta(ctx, xl, metaArr, errs)
+	if err != nil {
+		return err
+	}
+
+	for i, xlMeta := range metaArr {
+		// clean xlMeta.Meta of all tags, before updating the new tags
+		newMeta := removeTags(xlMeta.Meta)
```

Comment:
No reason to do this, single key update doesn't require removal just update xlMeta.Meta[xhttp.AmzObjectTagging] = tags